bicytok/figures/figure2.py

```diff
@@ -53,16 +53,18 @@ def makeFigure():
         2: (CITE_DF["CellType3"] == "Treg Naive"),  # Naive Tregs
     }
 
-    # Set off_target based on offTargState
     if offTargState in off_target_conditions:
-        off_target = off_target_conditions[offTargState].astype(int)
+        off_target_mask = off_target_conditions[offTargState]
     else:
         raise ValueError("Invalid offTargState value. Must be 0, 1, or 2.")
 
-    KL_values, EMD_values = KL_EMD_1D(markerDF, on_target, off_target)
+    on_target_values = markerDF[on_target.astype(bool)].values
+    off_target_values = markerDF[off_target_mask].values
 
-    top_5_KL_indices = np.argsort(KL_values)[-5:]
-    top_5_EMD_indices = np.argsort(EMD_values)[-5:]
+    KL_values, EMD_values = KL_EMD_1D(on_target_values, off_target_values)
```

Comment:
Sorry for the confusion. The on and off target arrays should just be arrays full of boolean values 
ie np.array([True, False, False...]). In this case you can just input off_target_mask and on_target without the .astype(int). Let me know if this doesn't make sense.